### PR TITLE
add new conf-bash package so we can depend on bash's availability

### PR DESCRIPTION
**Pull request by chetmurthy**
Hopefully this goes thru straight; I don't know how several of these distros name their bash package.

### CODE REVIEW

**Comment by kit-ty-kate** [comment]
Is this a duplicate of #22457 or do you have something else in mind?

**Comment by chetmurthy** [comment]
I didn't think to check for an open issue, sorry, I should have.  Reading thru the conversation and two of the linked-to PRs, it doesn't appear that conf-bash was actually introduced ?  Is it a primitive package (so doesn't need to be created) ?

I'd like to use conf-bash as a dependency (someone said that one of the packages I maintain wasn't building correctly on some platform, and this seems the best way to address it).

**Comment by kit-ty-kate** [comment]
Oops i meant to link towards the PR not the issue: https://github.com/ocaml/opam-repository/pull/22683

**Comment by chetmurthy** [comment]
Ah,  I see now.  This PR isnt' yet approved.  I also see that it only specifies depexts for `os-family = "bsd"`.

Should I just wait for this to go thru and then supply a patch to add other os-families ?

**Comment by raphael-proust** [comment]
@chetmurthy I've added alpine to #22683 because it's also not installed by default there. I don't know if we need all the distros because it is always there for a lot of them. I don't have a strong opinion on this.

I guess we can merge in either order. Idk.

By the way are there existing packages you are considering this for or is it for soon-to-be-released packages?

**Comment by chetmurthy** [comment]
I created this conf-package in response to a user who wanted me to switch to using `/bin/sh` for configure scripts (from `/bin/bash`), b/c it was breaking builds on some platform that the user did not specify.  I can only imagine what screwed-up platform it is, that has GCC, but not bash.

I figured might as well specify all the platforms, since it costs nothing and .... "suspenders and a belt".

**Comment by chetmurthy** [comment]
Sorry,  so yes, I plan to add this dependency to a bunch of packages that use bash for configuration.  But those packages already are released, so this isn't blocking anything.

**Comment by MisterDA** [comment]
Ran into this problem installing some packages on FreeBSD, they were using the `(bash …)` dune stanza, but bash isn't installed by default on this system.

**Comment by chetmurthy** [comment]
Is there any ETA on whether/when either this PR, or the other one ( https://github.com/ocaml/opam-repository/pull/22683 ) would be released ?  I see that the other one is still in a busted state ?

**Comment by kit-ty-kate** [comment]
https://github.com/ocaml/opam-repository/pull/22683 is now merged

File: packages/conf-bash/conf-bash.0.1/opam [hunk 1]
```diff
@@ -0,0 +1,32 @@
+opam-version: "2.0"
+maintainer: "bash-maintainers@gnu.org"
+homepage: "http://www.gnu.org/software/bash"
+authors: "https://tiswww.case.edu/php/chet/bash/bashtop.html#TOCMaintainer"
+bug-reports: "https://github.com/ocaml/opam-repository/issues"
+license: "GPL-3.0-only"
+build: [
+  ["which" "bash"]
+]
```
[the remark this answers is not in the document]
Comment (MisterDA) [reply packages/conf-bash/conf-bash.0.1/opam:9]: yes, we need to stop using `which`. It is unavailable on Windows generally, deprecated on Debian… See #18452 for some explanations.

**Pull request closed without merging**